stdint.h has C definitions which troubles assembler. and if asm/ptrace.h 
is included in assembler context, then it should not include the 
C-specific stdint.h file.
  This was causing glibc building issue on ppc.
Looks like this issue is introduced on the 2.6.34 kernel.

2010-06-024
Signed-Off-By:Nitin A Kamble <nitin.a.kamble@intel.com>
Acked-By:Jeff Dike <jdike@linux.intel.com>

Index: linux-2.6.34/arch/powerpc/include/asm/ptrace.h
===================================================================
--- linux-2.6.34.orig/arch/powerpc/include/asm/ptrace.h
+++ linux-2.6.34/arch/powerpc/include/asm/ptrace.h
@@ -27,8 +27,10 @@
 #ifdef __KERNEL__
 #include <linux/types.h>
 #else
+#ifndef __ASSEMBLY__
 #include <stdint.h>
 #endif
+#endif
 
 #ifndef __ASSEMBLY__
 
